docs/docs/launch-manual/sections/core-hacking/sections/publishing.md

```diff
@@ -0,0 +1,93 @@
+packages via the command line. This is invoked by using the `pulsar` command
+with the `-p` or `--package` option. The `pulsar -p` command can also be used to
+publish Pulsar packages to the public registry and update them.
+
+See more in [Using PPM](#using-ppm).
+
+### Prepare Your Package
+
+There are a few things you should double check before publishing:
+
+- Your `package.json` file has `name`, `description`, and `repository` fields.
+- Your `package.json` file has a `version` field with a value of `"0.0.0"`.
+- Your `package.json` file has an `engines` field that contains an entry for
+  `atom` such as: `"engines": {"atom": ">=1.0.0 <2.0.0"}`.
+- Your package has a `README.md` file at the root.
+- Your `repository` URL in the `package.json` file is the same as the URL of
+  your repository.
+- Your package is in a Git repository that has been pushed to
+  [GitHub](https://github.com). Follow [this guide](https://help.github.com/articles/importing-a-git-repository-using-the-command-line/)
+  if your package isn't already on GitHub.
+
+### Publish Your Package
+
+Before you publish a package it is a good idea to check ahead of time if a
+package with the same name has already been published to
+[the Pulsar Package Repository](https://web.pulsar-edit.dev/packages). You can
+do that by visiting `https://web.pulsar-edit.dev/packages/your-package-name` to
+see if the package already exists. If it does, update your package's name to
+something that is available before proceeding.
+
+Now let's review what the `pulsar -p publish` command does:
+
+1. Registers the package name on Pulsar Package Repository if it is being
+   published for the first time.
+2. Updates the `version` field in the `package.json` file and commits it.
+3. Creates a new [Git tag](https://git-scm.com/book/en/Git-Basics-Tagging) for
+   the version being published.
+4. Pushes the tag and current branch up to GitHub.
+5. Updates Pulsar Package Repository with the new version being published.
+
+Now run the following commands to publish your package:
+
+```sh
+$ cd path-to-your-package
+$ pulsar -p publish minor
+```
+
+If this is the first package you are publishing, the `pulsar -p publish` command
+may prompt you for your GitHub username and password. If you have two-factor
+authentication enabled, use a [personal access token](https://help.github.com/articles/creating-a-personal-access-token-for-the-command-line/)
+in lieu of a password. This is required to publish and you only need to enter
+this information the first time you publish. The credentials are stored securely
+in your [keychain](<https://en.wikipedia.org/wiki/Keychain_(software)>) once you
+login.
+
+Your package is now published and available on Pulsar Package Repository. Head
+on over to `https://web.pulsar-edit.dev/packages/your-package-name` to see your
+package's page.
+
+With `pulsar -p publish`, you can bump the version and publish by using
+
+```sh
+$ pulsar -p publish <version-type>
+```
+
```

Comment:
Feel like this whole part could just be a link to `semver.org`, but what do you think?

Reply:
I want to say yes but at the same time I want to say no. Basically I think our goal should be to make this process as easy as possible for people to understand, really keep that barrier for entry as low as possible. The semver website isn't necessarily the easiest to understand on a casual read but then again I assume most people making packages have been exposed to these concepts in the past...

Maybe we could do a best of both worlds? Something like including the summary from semver + a link plus a single example:

So

> ... where `version-type` can be `major`, `minor` and `patch`.
>   MAJOR version when you make incompatible API changes
    MINOR version when you add functionality in a backwards compatible manner
    PATCH version when you make backwards compatible bug fixes
e.g. to bump `myPackage` from 1.**0**.0 to 1.**1**.0:
`$ pulsar -p publish minor`
> Check out [semantic versioning](https://semver.org) to learn more about best practices for versioning your package releases.

Reply:
Oh yeah @Daeraxa this seems like the best thing.

Totally agree we want it to be easy for new comers to the field, and this seems like it makes it easy. While taking some of that responsibility off of us.

Reply:
I've added this now if you want to check.